desktop_version/src/DiscordNetwork.c

```diff
@@ -0,0 +1,188 @@
+    if (libHandle == NULL)
+    {
+        vlog_info(DISCORD_LIBRARY " not found!");
+        vlog_debug("Can't load object %s : %s\n", DISCORD_LIBRARY, SDL_GetError());
+        return 0;
+    }
+
+#define FOREACH_FUNC(rettype, name, params) \
+    name = (rettype (*) params) (intptr_t) SDL_LoadFunction(libHandle, #name); \
+    if (!name) \
+    { \
+        vlog_error(DISCORD_LIBRARY " symbol " #name " not found!"); \
+        ClearPointers(); \
+        return 0; \
+    }
+    FUNC_LIST
+#undef FOREACH_FUNC
+    SDL_memset(&app, 0, sizeof(app));
+
+    struct DiscordCreateParams params;
+    params.client_id = DISCORD_CLIENT_ID;
+    params.flags = DiscordCreateFlags_NoRequireDiscord;
+
+    if (!DISCORD_REQUIRE(DiscordCreate(DISCORD_VERSION, &params, &app.core)))
+    { // Discord's API couldn't initialise, so just ignore it
+        discordPostInit = true; // even if it fails, set post init to true.
+        vlog_debug("Discord API failed to initialise!");
+        discordDetected = false;
+        return 0;
+    }
+
+    if (app.core == NULL)
+    {
+        discordPostInit = true;
+        vlog_debug("app.core == null. DiscordCreate failed with a positive result?\nCheck DISCORD_REQUIRE  for bugs.");
+        discordDetected = false;
+        return 0;
+    }
+    discordPostInit = true;
+
+    // Placeholder, is this really nesaccary
+    SDL_strlcpy(activity.assets.large_image, "vvvvvv", sizeof(activity.assets.large_image));
+    SDL_strlcpy(activity.assets.large_text, "Outside Dimension VVVVVV", sizeof(activity.assets.large_text));
+
+    app.activityMan = app.core->get_activity_manager(app.core);
+    app.activityMan->update_activity(app.activityMan, &activity, NULL, NULL);
+
+
+    discordDetected = true;
+    return 1;
+}
+
+int32_t DISCORD_update(void)
+{
+    if (libHandle == NULL)
+    {
+        // no Discord or just shutdown
+        return 0;
+    }
+    if (app.core == NULL || !discordDetected)
+    {
+        // No Discord
+        return 0;
+    }
+
+    if (!DISCORD_REQUIRE(app.core->run_callbacks(app.core)))
+    {
+        // Something or other is wrong, but do we care?
+        return 0;
+    }
+    return 1;
+}
+void DISCORD_unlockAchievement(const char* name)
+{
+    // No "achievements" in Discord
+}
+
+void DISCORD_setRPC(const char* area, const char* roomname)
+{
+    if (app.activityMan == NULL)
+    {
+     	vlog_debug("No activityMan!  - it\'s fine, we can recreate this" );
+        app.activityMan = app.core->get_activity_manager(app.core);
+    }
+    if (SDL_strcmp(activity.state, roomname) != 0 || SDL_strcmp(activity.assets.large_text, area) != 0)
+    {
+     	SDL_strlcpy(activity.state, roomname, sizeof(activity.state));
+        SDL_strlcpy(activity.assets.large_image, "vvvvvv", sizeof(activity.assets.large_image));
+        SDL_strlcpy(activity.assets.large_text, area, sizeof(activity.assets.large_text));
+
+        app.activityMan->update_activity(app.activityMan, &activity, NULL, NULL);
+    }
+
```

Comment:
Newlines in this file seem generally inconsistent. This should probably be cleaned up.